### ISSUE TITLE
Testflight not receiving updates

### ISSUE THREAD

**Issue by JanithaR** [microsoft/react-native-code-push#1714]
I have an app which essentially has 3 variants published as 2 different apps.

1. Beta (app 1)
2. PreRelease (app 2)
3. Production (app 2)

Beta connects to the test server and only published as a Testflight app and on Android internal track.

PreRelease connects to the production server and only published as a Testflight app and on Android internal track. (This is so that we are sure everything that's about to go to end-users are 100% working both on the mobile and server-side.)

Production connects to the production server and what's being used by the end-users. (available on stores)

I'm using react-native-config to manage my environment configurations and for this, I use 2 Android flavours and 3 iOS schemes. Only 2 Android flavours because I can pass which .env file to use when building.

Essentially the only difference between PreRelease and Production is the Codepush key. They both refer to the same Codepush app, only, PreRlease is Staging and Production is Production.

I hope my setup is clear. :)

### Problem
When I push an update to my Beta;

- Android receives it
- iOS app running on either Beta.Debug or Beta.Release config receives it
- Testers who have installed the app via Testflight **DON'T** receive it

All 3 version is having the same binary version.

### Expected Behavior
Testers who have the app installed the app via Testflight should also receive the update.

### Actual Behavior
Log from the Testflight app

> [CodePush] Loading JS bundle from file:///var/containers/Bundle/Application/68D8AF5B-D109-4524-95D7-D7706444937F/Facilit%20FDVU.app/main.jsbundle

Log from the simulator

> [CodePush] Loading JS bundle from file:///Users/janithar/Library/Developer/CoreSimulator/Devices/2BC36BD5-6557-49EF-A693-40A4336FCB4B/data/Containers/Data/Application/C824924C-A1F8-4286-A45C-C3D4ADE61653/Library/Application<decode: mismatch for [%20S] got [STRING private sz:332]>upport/CodePush/23d97f583b4239618b60936efb72a7af6f79826ef5234d98b47f7e0efc7a7b39/CodePush/main.jsbundle

I suspect it may have to do something with,
```
- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [CodePush bundleURL];
#endif
}
```
but cluless as to what could be going wrong.

### Environment

* react-native-code-push version:^5.7.0
* react-native version:0.59.10

**Comment by JanithaR** [comment]
After doing several hours of debugging I noticed that the manually Archived and installed builds from XCode works as expected. So that narrows the issue further down to the build released via Testflight.

**Comment by JanithaR** [comment]
I'm closing this as I have found out that when Archives are made on Circle CI, somehow my vars are not picked up by Info.plist.

**Comment by AJPicard913** [comment]
Hey @JanithaR  we're having the same issue. What did you do to solve it?

**Comment by JanithaR** [comment]
I just omitted to put the Codepush keys in the .env files and followed these steps.
https://github.com/microsoft/react-native-code-push/blob/master/docs/setup-ios.md

There is an issue where "key/values" are not available to the Info.plist file when using react-native-config. There are workarounds but I recommend to just keep the Codepush keys away from it until the issue is sorted out by the maintainers of react-native-config.

You will be able to find related issues in that repo.

**Comment by Acousticdesk** [comment]
I've faced the same issue, the reason was due to I have my appcenter account registered using a different email to the one in the TestFlight, so I just haven't had enough permissions to push new builds to testflight.

However, no errors occurred for both - appcenter/testflight that made me spend extra time resolving this issue.

Hope it helps someone!

**Comment by sufian1999** [comment]
"react": "17.0.2",
"react-native": "0.68.2",
"react-native-code-push": "^7.1.0"

I pushed code with this:
appcenter codepush release-react -a owner-email/app-name -d Production -m --disabled

When the 'Just Now' changes to the Date Time. I Enabled the release and when I open the TestFlight, it downloads the update after 1-2 minutes and changes reflect on the next restart.

Today, I tested it 2 times on my TestFlight, it's working.

Suppose any issue comes up once my app is uploaded via the App Store. I will let you people know.
